src/analyses/variable-sensitivity/variable_sensitivity_object_factory.cpp

```diff
@@ -10,8 +10,217 @@
+  auto choices = std::string("");
+  for(auto &kv : mapping)
+  {
+    choices += (!choices.empty() ? "|" : "");
+    choices += kv.first;
+  }
+
+  return invalid_command_line_argument_exceptiont{
+    "Unknown argument '" + bad_argument + "'", option, option + " " + choices};
+}
+
+ABSTRACT_OBJECT_TYPET vsd_configt::option_to_abstract_type(
+  const optionst &options,
+  const std::string &option_name,
+  const option_mappingt &mapping,
+  ABSTRACT_OBJECT_TYPET default_type)
+{
+  const auto argument = options.get_option(option_name);
+
+  if(argument.empty())
+    return default_type;
+
+  auto selected = mapping.find(argument);
+  if(selected == mapping.end())
+  {
+    throw invalid_argument(option_name, argument, mapping);
+  }
+  return selected->second;
+}
+
+template <class abstract_object_classt, class context_classt>
+abstract_object_pointert initialize_context_abstract_object(
+  const typet type,
+  bool top,
+  bool bottom,
+  const exprt &e,
+  const abstract_environmentt &environment,
+  const namespacet &ns)
+{
+  if(top || bottom)
+  {
+    return abstract_object_pointert(new context_classt(
+      abstract_object_pointert(new abstract_object_classt(type, top, bottom)),
+      type,
+      top,
+      bottom));
+  }
+  else
+  {
+    PRECONDITION(type == ns.follow(e.type()));
+    return abstract_object_pointert(new context_classt(
+      abstract_object_pointert(new abstract_object_classt(e, environment, ns)),
+      e,
+      environment,
+      ns));
+  }
+}
+
+/// Function: variable_sensitivity_object_factoryt::initialize_abstract_object
+/// Initialize the abstract object class and return it.
+///
+/// \param type: the type of the variable
+/// \param top: whether the abstract object should be top in the
+///             two-value domain
+/// \param bottom: whether the abstract object should be bottom in the
+///                two-value domain
+/// \param e: if top and bottom are false this expression is used as the
+///           starting pointer for the abstract object
+/// \param environment: the current abstract environment
+/// \param ns: namespace, used when following the input type
+///
+/// \return An abstract object of the appropriate type.
+///
+template <class abstract_object_classt>
+abstract_object_pointert initialize_abstract_object(
+  const typet type,
+  bool top,
+  bool bottom,
+  const exprt &e,
+  const abstract_environmentt &environment,
+  const namespacet &ns,
+  const vsd_configt &configuration)
+{
+  if(configuration.context_tracking.data_dependency_context)
+    return initialize_context_abstract_object<
+      abstract_object_classt,
+      data_dependency_contextt>(type, top, bottom, e, environment, ns);
+  if(configuration.context_tracking.last_write_context)
+    return initialize_context_abstract_object<
+      abstract_object_classt,
+      write_location_contextt>(type, top, bottom, e, environment, ns);
+  else
+  {
+    if(top || bottom)
+    {
+      return abstract_object_pointert(
```

Comment:
Nit picking (here and elsewhere): use curly braces for constructors as indicated in the coding guidelines.